### ISSUE TITLE
App icon not updated in Task View (Win+Tab) or Task Switcher (Alt+Tab)

### ISSUE THREAD

**Issue by jess-rod** [microsoft/microsoft-ui-xaml#6104]
### Describe the bug

Despite having customized all the app icons (for all scales, unplated, light unplated...), the icon that shows when pressing Alt+Tab or Windows+Tab is the default (not the customized) icon.

The customized icon does show on the taskbar, start menu (tiles)... 

[Docs reference.](https://docs.microsoft.com/en-us/windows/apps/design/style/app-icons-and-logos?ranMID=46131&ranEAID=a1LgFw09t88&ranSiteID=a1LgFw09t88-g2RQ7ic6gPG6Xx0zw14TFg&epi=a1LgFw09t88-g2RQ7ic6gPG6Xx0zw14TFg&irgwc=1&OCID=AID2200057_aff_7806_1243925&tduid=%28ir__ucmrpk6gekkf6whl2qniaboutn2xrbotmoa1319i00%29%287806%29%281243925%29%28a1LgFw09t88-g2RQ7ic6gPG6Xx0zw14TFg%29%28%29&irclickid=_ucmrpk6gekkf6whl2qniaboutn2xrbotmoa1319i00#:~:text=The%20app%20icon%20asset%20appears%20in%20a%20lot%20of%20places%3A%20the%20Windows%20taskbar%2C%20the%20task%20view%2C%20ALT%2BTAB%2C%20and%20the%20lower-right%20corner%20of%20Start%20tiles)

### Steps to reproduce the bug

1. Create a WinUI desktop app.
2. Generate all the assets using Visual Studio Asset Generator.
3. Debug the app (F5), or deploy the app, the result is the same.
4. Press Alt+Tab (or Windows+Tab).
5. See the default icon.

### Expected behavior

5. See the customized icon.

### NuGet package version

WinUI 3 - Windows App SDK 0.8: 0.8.0 (If you're seeing your issue in older previews of WinUI 3, please try this release)

### Windows app type

- [ ] UWP
- [X] Win32

### Device form factor

Desktop

### Windows version

October 2020 Update (19042)

**Comment by pratikone** [comment]
I tried it out and can confirm it. Thanks for posting this. @gabbybilka @MikeHillberg we can explore behavior when a user sets app icon whether to display it as window icon or not.

**Comment by stevenbrix** [comment]
almost every windows app in the world shows the app icon here, it would be nice to have the default behavior do that

**Comment by agat366** [comment]
Yeah, I am just laughing out again and again about how Microsoft is "**attentive**" to its own "native" (yep, in quotes) system! That's just ridiculous how buggy and hard for implementation are many primitive features of "native" Windows platform for, again "native" (once again in quotes) developers.
I guess, this is really some sort of Conspiracy Theory which states that Microsoft _tries to get rid of its native devs_ to migrate them to Apple or other mobile and desktop platforms. **For years**! They are like whispering to us: "Guys, please use our perfect cross-platform MAUI, but don't even think to release them for Windows! Don't waste your lives! iOS, MacOS, maybe some Linux... They are much cooler and more reliable, but please, please, I am begging you, just forget Windows!" lol
... please! )))

**Comment by AkilanRagavaswamy** [comment]
Ohh.. Still there is no update... 

![Image](https://github.com/user-attachments/assets/16b559ce-87d4-4830-9950-b51d6a24cc7e)

Just sharing a work around approach for this issue, we can set the icon using the **SetIcon** method from **AppWindow**. 

Steps:
1. Add the icon file to assets, make sure it is in .ico format.
2. Right click -> Properties -> Change the below properties, 
        -> Change Build Action to "Content"
        -> Change Copy to Output Directory to "Copy Always" or "Copy if Newer"
3. Invoke the **SetIcon** method from **AppWindow** in the **OnLaunched** method of **App.xaml.cs**. Refer the snippet below for sample,

`protected override async void OnLaunched(Microsoft.UI.Xaml.LaunchActivatedEventArgs args) 
{
    Window = new MainWindow();
    string iconPath = Path.Combine(AppContext.BaseDirectory, "Assets/WinUI.ico"); 
    Window.AppWindow.SetIcon(iconPath);
    Window.Activate();
}`

Hope this helps!!!

**Comment by niels9001** [comment]
I think this is related.. or might actually be a duplicate: #11023. The caveat there is that it does show up under debug, but not always in Release mode? Or when packaged through the Store?

We are running into this too with PowerToys
